include/boost/iterator/is_iterator.hpp

```diff
@@ -68,10 +66,10 @@ struct is_iterator_impl :
 
 template< typename T >
 struct is_iterator_impl< T* > :
-    public boost::conjunction<
-        boost::is_complete< T >,
-        boost::negation< boost::is_function< T > >
-    >::type
+    std::integral_constant<
+        bool
+      , boost::is_complete<T>::value && !std::is_function<T>::value
```

Comment:
`boost::conjunction` is more optimal. I think, it should stay or at least be abstracted like in other libraries, e.g. like [here](https://github.com/boostorg/scope/blob/eea343a918c586387b1b717c31c0a24bc612e5c9/include/boost/scope/detail/type_traits/conjunction.hpp).

Also, trailing commas are more preferable than leading commas.